contracts/PriceOracle/libs/PriceGroupsLib.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {Groups, PriceGroup} from "./IPriceGroupsLib.sol";
+
+/**
+ * @title PriceGroupsLib Library
+ * @author Samuil Borisov
+ * @notice Library for inserting a validator with a price in groups, based on the price consensus
+ */
+library PriceGroupsLib {
+    /**
+     * @notice Inserts a new validator with a price in a group
+     * @param self The groups to insert the validator in
+     * @param validator The validator to insert
+     * @param price The price to insert
+     * @dev Creates groups, where validators whose prices are within 1% of the average price are grouped together
+     */
+    function insert(Groups storage self, address validator, uint256 price) internal {
+        // assert(price != 0);  // This is not needed, as the price is already checked in the calling function
+        // assert(validator != address(0)); // This is not needed, as the validator is the msg.sender, who should always be a valid address
+        /// @dev We cannot check if the validator is already in the group, so make sure to check if validator already voted this before calling the function (in our case, this is done in the vote function)
```

Comment:
move this comment above the function declaration so it will be added to the NatSpec generated docs